data_augmentation.py

```diff
@@ -223,6 +220,15 @@ def get_augmenter_pipeline(strength_enum):
 
     return aug_pipeline
 
+def flipRL(image, keypoints):
+    if random.random() < RL_FLIP:
+        return image, keypoints
+    flip = iaa.Fliplr(1.0)
+    flipped_img, flipped_kps = flip(image=image, keypoints=keypoints)
+    flat_map = lambda f, xs: [y for ys in xs for y in f(ys)] # https://dev.to/turbaszek/flat-map-in-python-3g98
+    flipped_fixed_kps = [flipped_kps[0]] + flat_map(lambda pair : (pair[1],pair[0]), zip(flipped_kps[1::2],flipped_kps[2::2]))
+    # Leave nose unchanged, flip every pair of eye, ear, etc...
+    return flipped_img, flipped_fixed_kps
```

Comment:
This looks great to me good catch! Just to confirm you're flipping it so that it remains in the keypoints order in constants?

Reply:
Yes, so:
`["nose","left_eye","right_eye","left_ear","right_ear","left_shoulder","right_shoulder","left_elbow","right_elbow","left_wrist","right_wrist","left_hip","right_hip","left_knee","right_knee","left_ankle","right_ankle"]`
becomes:
`["nose","right_eye","left_eye","right_ear","left_ear"...]`

This is necessary b/c when you mirror flip an image, a person's right side becomes their left side, so after fixing the coords, we need to swap labels with the other side.

Reply:
Ahh gotcha I wonder how this would work if we feed an augmented image into OKS or PCK if we can't directly compare to the ground truth.

Reply:
We aren't applying augmentation to our validation or test set (and i'm not sure if we should). If we do, we can hold on to the augmented ground truth keypoints and use that as our ground truth. We will have to avoid running PCK on the training generator without accounting for this though

Reply:
We aren't applying augmentation to our validation or test set (and i'm not sure if we should). If we do, we can hold on to the augmented ground truth keypoints and use that as our ground truth. We will have to avoid running PCK on the training generator without accounting for this though